sdk/iothub/src/service/mod.rs

```diff
@@ -403,6 +407,105 @@ impl ServiceClient {
     {
         DesiredTwinBuilder::new(&self, device_id.into(), None, Method::PUT)
     }
+
+    /// Get the identity of a given device
+    ///
+    /// ```
+    /// use iothub::service::ServiceClient;
+    /// use serde_json;
+    ///
+    /// # let connection_string = "HostName=cool-iot-hub.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=YSB2ZXJ5IHNlY3VyZSBrZXkgaXMgaW1wb3J0YW50Cg==";
+    /// let iothub = ServiceClient::from_connection_string(connection_string, 3600).expect("Failed to create the ServiceClient!");
+    /// let device = iothub.get_device_identity("some-device");
+    /// ```
+    pub async fn get_device_identity<S>(&self, device_id: S) -> Result<Device, AzureError>
```

Comment:
These don't really match the pattern of request builders we've been using in other crates. That might be fine, but we probably need to think more deeply about if the iothub crate should follow the same pattern.

Reply:
I'd be happy to change it so it's similar to the other crates, no problem at all. I did notice there is a "common" http client within the core crate that was added a while ago. Is this what you mean that should be used instead? Could you link me an example of the ideal way we'd be doing requests?